packages/opentelemetry-instrumentation-openai/opentelemetry/instrumentation/openai/shared/__init__.py

```diff
@@ -129,6 +131,13 @@ def _set_response_attributes(span, response):
         )
 
 
+def _get_openai_base_url():
```

Comment:
:question: Is this function still always returning empty string?

I can confirm this is behavior if no initialization is done.
```python
In [2]: def _get_openai_base_url():
   ...:     base_url = openai.base_url if hasattr(openai, "base_url") else openai.api_base
   ...:     if not base_url:
   ...:         return ""
   ...:     return base_url
   ...: 

In [3]: import openai

In [4]: _get_openai_base_url()
Out[4]: ''

In [5]: 
```

On the other hand, if you inspect the client instance, you should get the base url:

```python
In [6]: client = openai.OpenAI()

In [7]: client.base_url
Out[7]: URL('https://api.openai.com/v1/')
```

Does this help you?

Reply:
I've reviewed your code, sadly I don't see an easy way to extract this URL from an instance in the current instrumentation code. 

My only idea was to instrument the client constructor so you can inject a listener which retrieves the URL for you. If you store it in a Singleton, you could retrieve it while calling this function:

```
def _handle_response(response, span, token_counter=None, choice_counter=None, duration_histogram=None, duration=None):
    if is_openai_v1():
        response_dict = model_as_dict(response)
    else:
        response_dict = response

    # metrics record
    _set_chat_metrics(token_counter, choice_counter, duration_histogram, response_dict, duration)

    # span attributes
    _set_response_attributes(span, response_dict)

    if should_send_prompts():
        _set_completions(span, response_dict.get("choices"))

    return response
```

Or maybe even directly inside the `_set_chat_metrics`.

But again, quite some effort for just one URL :)

Reply:
@nirga what do you think? Should we track this URL as a separate issue?

Reply:
@paolorechia not sure I follow - I think this function does return the right base URL, no?

Reply:
Yes, @Humbertzhang mentioned in a comment, he couldn’t get this URL to work correctly, which why I looked a bit on that

Reply:
@Humbertzhang I think this is the logic we should use to get it: https://github.com/traceloop/openllmetry/blob/main/packages/opentelemetry-instrumentation-openai/opentelemetry/instrumentation/openai/shared/__init__.py
(depending on the SDK version, it will be in different attributes).

Reply:
For v1, it should be `instance._client.base_url`, and for v0 it should be the code you wrote below